include/nbl/builtin/hlsl/glsl_compat/core.hlsl

```diff
@@ -56,11 +56,11 @@ T atomicCompSwap(NBL_REF_ARG(T) ptr, T comparator, T value)
 }
 
 void barrier() {
-    spirv::controlBarrier(2, 2, 0x8 | 0x100);
+    spirv::controlBarrier(spv::ScopeWorkgroup, spv::ScopeWorkgroup, spv::MemorySemanticsAcquireReleaseMask | spv::MemorySemanticsWorkgroupMemoryMask);
```

Comment:
btw we might have to alias to HLSL control barrier unless we can detect tesselation control vs compute shader, because GLSL spec says
```
barrier() (compute) -> OpControlBarrier(/*Execution*/Workgroup,
/*Memory*/Workgroup,
/*Semantics*/AcquireRelease |
WorkgroupMemory)
barrier() (tess control) -> OpControlBarrier(/*Execution*/Workgroup,
/*Memory*/Invocation,
/*Semantics*/None)
```